cmd/argo/commands/list.go

```diff
@@ -21,117 +22,110 @@ import (
-			namespace := client.Namespace()
-			if listArgs.allNamespaces {
-				namespace = ""
-			}
-
-			var tmpWorkFlows []wfv1.Workflow
-			for {
-				log.WithField("listOpts", listOpts).Debug()
-				wfList, err := serviceClient.ListWorkflows(ctx, &workflowpkg.WorkflowListRequest{Namespace: namespace, ListOptions: listOpts})
-				errors.CheckError(err)
-				tmpWorkFlows = append(tmpWorkFlows, wfList.Items...)
-				if wfList.Continue == "" {
-					break
-				}
-				listOpts.Continue = wfList.Continue
-			}
-
-			var tmpWorkFlowsSelected []wfv1.Workflow
-			if listArgs.prefix == "" {
-				tmpWorkFlowsSelected = tmpWorkFlows
-			} else {
-				tmpWorkFlowsSelected = make([]wfv1.Workflow, 0)
-				for _, wf := range tmpWorkFlows {
-					if strings.HasPrefix(wf.ObjectMeta.Name, listArgs.prefix) {
-						tmpWorkFlowsSelected = append(tmpWorkFlowsSelected, wf)
-					}
-				}
+			if !allNamespaces {
+				listArgs.namespace = client.Namespace()
 			}
-
-			var workflows wfv1.Workflows
-			if listArgs.since == "" {
-				workflows = tmpWorkFlowsSelected
-			} else {
-				workflows = make(wfv1.Workflows, 0)
-				minTime, err := argotime.ParseSince(listArgs.since)
-				errors.CheckError(err)
-				for _, wf := range tmpWorkFlowsSelected {
-					if wf.Status.FinishedAt.IsZero() || wf.ObjectMeta.CreationTimestamp.After(*minTime) {
-						workflows = append(workflows, wf)
-					}
-				}
-			}
-			sort.Sort(workflows)
-			err := printer.PrintWorkflows(workflows, os.Stdout, printer.PrintOpts{
+			workflows, err := listWorkflows(ctx, serviceClient, listArgs)
+			errors.CheckError(err)
+			err = printer.PrintWorkflows(workflows, os.Stdout, printer.PrintOpts{
 				NoHeaders: listArgs.noHeaders,
-				Namespace: listArgs.allNamespaces,
+				Namespace: allNamespaces,
 				Output:    listArgs.output,
 			})
 			errors.CheckError(err)
 		},
 	}
-	command.Flags().BoolVar(&listArgs.allNamespaces, "all-namespaces", false, "Show workflows from all namespaces")
+	command.Flags().BoolVar(&allNamespaces, "all-namespaces", false, "Show workflows from all namespaces")
 	command.Flags().StringVar(&listArgs.prefix, "prefix", "", "Filter workflows by prefix")
+	command.Flags().StringVar(&listArgs.finisheAfter, "older", "", "List completed workflows finished before the specified duration (e.g. 10m, 3h, 1d)")
 	command.Flags().StringSliceVar(&listArgs.status, "status", []string{}, "Filter by status (comma separated)")
 	command.Flags().BoolVar(&listArgs.completed, "completed", false, "Show only completed workflows")
 	command.Flags().BoolVar(&listArgs.running, "running", false, "Show only running workflows")
 	command.Flags().StringVarP(&listArgs.output, "output", "o", "", "Output format. One of: wide|name")
-	command.Flags().StringVar(&listArgs.since, "since", "", "Show only workflows newer than a relative duration")
+	command.Flags().StringVar(&listArgs.createdSince, "since", "", "Show only workflows created after than a relative duration")
```

Comment:
is it the same as older?

Reply:
not quite `--older` had really weird implementation: 
```
workflows = make(wfv1.Workflows, 0)
				minTime, err := argotime.ParseSince(listArgs.since)
				errors.CheckError(err)
				for _, wf := range tmpWorkFlowsSelected {
					if wf.Status.FinishedAt.IsZero() || wf.ObjectMeta.CreationTimestamp.After(*minTime) {
						workflows = append(workflows, wf)
					}
				}
```
Why do we guard the creation after on finished at?